Solaris grep supports limited regular expressions.
Use GNU grep in mellon_create_metadata.sh script.

diff -ru mod_auth_mellon-0.14.1/mellon_create_metadata.sh mod_auth_mellon-0.14.1.new/mellon_create_metadata.sh
--- mod_auth_mellon-0.14.1/mellon_create_metadata.sh	2019-03-20 21:37:40.193318280 +0100
+++ mod_auth_mellon-0.14.1.new/mellon_create_metadata.sh	2019-03-20 22:00:57.596300418 +0100
@@ -30,7 +28,7 @@
     exit 1
 fi
 
-if ! echo "$BASEURL" | grep -q '^https\?://'; then
+if ! echo "$BASEURL" | /usr/bin/ggrep -q '^https\?://'; then
     echo "$PROG: The URL must start with \"http://\" or \"https://\"." >&2
     exit 1
 fi
@@ -71,7 +69,7 @@
 
 rm -f "$TEMPLATEFILE"
 
-CERT="$(grep -v '^-----' "$OUTFILE.cert")"
+CERT="$(/usr/bin/ggrep -v '^-----' "$OUTFILE.cert")"
 
 cat >"$OUTFILE.xml" <<EOF
 <EntityDescriptor entityID="$ENTITYID" xmlns="urn:oasis:names:tc:SAML:2.0:metadata" xmlns:ds="http://www.w3.org/2000/09/xmldsig#">
